CMakeLists.txt

```diff
@@ -51,58 +66,87 @@ if(APPLE)
 	if(DSYMUTIL_PROGRAM)
 		foreach(lang C CXX)
 			foreach(var LINK_EXECUTABLE CREATE_SHARED_LIBRARY)
-				set(CMAKE_${lang}_${var} "${CMAKE_${lang}_${var}}" "${DSYMUTIL_PROGRAM} <TARGET>")
+				#set(CMAKE_${lang}_${var} "${CMAKE_${lang}_${var}}" "${DSYMUTIL_PROGRAM} <TARGET>")
 			endforeach()
 		endforeach()
 	endif()
 endif()
 
-
-OPTION(WITH_ASAN_OPTION "Build sentry-native with address sanitizer" OFF)
+option(WITH_ASAN_OPTION "Build sentry-native with address sanitizer" OFF)
 if(WITH_ASAN_OPTION)
-	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -fsanitize=address -fno-omit-frame-pointer")
-	link_libraries("-fsanitize=address")
+	add_compile_options(-g -fsanitize=address -fno-omit-frame-pointer)
+	link_libraries(-fsanitize=address)
 endif()
 
-OPTION(WITH_TSAN_OPTION "Build sentry-native with thread sanitizer" OFF)
+option(WITH_TSAN_OPTION "Build sentry-native with thread sanitizer" OFF)
 if(WITH_TSAN_OPTION)
-	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -fsanitize=thread -fno-omit-frame-pointer")
-	link_libraries("-fsanitize=thread")
+	add_compile_options(-g -fsanitize=thread -fno-omit-frame-pointer)
+	link_libraries(-fsanitize=thread)
 endif()
 
+option(SENTRY_CURL_SUPPORT "Buils sentry-native with curl support" ON)
+
+# builtin `list(TRANSFORM)` (https://cmake.org/cmake/help/latest/command/list.html#transform)
+# was only added in cmake 3.12, but we target 3.10, so we define a custom
+# functions which can prepend a prefix to all list elements.
+# copied from https://stackoverflow.com/a/27630120
+function(PREPEND var prefix)
+	set(listVar "")
+	foreach(f ${ARGN})
+		list(APPEND listVar "${prefix}/${f}")
+	endforeach(f)
+	set(${var} "${listVar}" PARENT_SCOPE)
+endfunction()
+
+# ===== sentry library =====
+
+add_library(sentry "${PROJECT_SOURCE_DIR}/vendor/mpack.c")
+add_library(sentry::sentry ALIAS sentry)
+add_subdirectory(src)
+set_target_properties(sentry PROPERTIES PUBLIC_HEADER "include/sentry.h")
+
+# https://gitlab.kitware.com/cmake/cmake/issues/18393
 if(BUILD_SHARED_LIBS)
-	add_definitions(-DSENTRY_BUILD_SHARED)
-else()
-	add_definitions(-DSENTRY_BUILD_STATIC)
+	if(APPLE)
+		sentry_install(FILES "$<TARGET_FILE:sentry>.dSYM" DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+	elseif(MSVC)
+		sentry_install(FILES "$<$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>:$<TARGET_PDB_FILE:sentry>>"
+			DESTINATION "${CMAKE_INSTALL_BINDIR}")
+	endif()
 endif()
 
-FIND_PACKAGE(CURL)
-if(CURL_FOUND)
```

Comment:
I think you should keep this `if`, and just override the `SENTRY_CURL_SUPPORT` var with the result.
CI fails on platforms that don’t have curl available.

Reply:
My opinion on this matter is that it's better to fail explicitly when a desired configuration cannot be set.
By failing, you know the binary will be as configured.
By overriding, you have to check the logs whether CURL has been found.
Maybe those platforms that do not have curl, should set `SENTRY_CURL_SUPPORT=OFF`?

Reply:
I disabled CURL in my latest batch of patches.

Reply:
@Swatinem
I present some alternatives:
- only enable curl on those platforms where you know CURL is installed (such as `sys.platform == "darwin"`)?
- use a package manager to install curl
e.g. conan:


<details>
<summary>Build using conan</summary>

Create a `conanfile.py` file in the root of `sentry-native` with the following contents:
```python
from conans import ConanFile, CMake, tools
from conans.errors import ConanInvalidConfiguration


class F(ConanFile):
    settings = "os", "arch", "compiler", "build_type"
    options = {
        "shared": [True, False],
        "fPIC": [True, False],
        "backend": ["default", "none", "crashpad", "inproc"],
        "with_curl": [True, False],
    }
    default_options = {
        "shared": False,
        "fPIC": True,
        "backend": "default",
        "with_curl": True,
    }

    generators = "cmake_find_package"

    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC

    def configure(self):
        if self.options.shared:
            del self.options.fPIC
        if self.options.backend == "default":
            if self.settings.os == "Windows" or tools.is_apple_os(self.settings.os):
                self.options.backend = "crashpad"
            elif self.settings.os == "Linux":
                self.options.backend = "inproc"
            else:
                raise ConanInvalidConfiguration("Unsupported os")
        if self.settings.os == "Linux" and self.options.backend == "crashpad":
            raise ConanInvalidConfiguration("crashpad backend is unsupported on Linux")
        elif self.settings.os == "Windows" and self.options.backend == "inproc":
            raise ConanInvalidConfiguration("inproc backend is unsupported on Windows")

    def requirements(self):
        if self.options.with_curl:
            self.requires("libcurl/7.68.0")

    def build(self):
        cmake = CMake(self)
        cmake.definitions["SENTRY_CURL_SUPPORT"] = self.options.with_curl
        cmake.definitions["SENTRY_BACKEND"] = str(self.options.backend).lower()
        cmake.configure()
        cmake.build()

    def package(self):
        cmake = CMake(self)
        cmake.install()

    def test(self):
        cmake = CMake()
        cmake.test()
```

and run the following commands:
```sh
python -m pip install --user conan
mkdir build
cd build
python -m conans.conan install ..
python -m conans.conan build ..
```

</details>